client/python/gradio_client/client.py

```diff
@@ -785,8 +785,8 @@ def serialize(self, *data) -> tuple:
             if t in ["file", "uploadbutton"]
         ]
         uploaded_files = self._upload(files)
-        self._add_uploaded_files_to_data(uploaded_files, list(data))
-
+        data = list(data)
```

Comment:
There was a bug here where the uploaded file info was actually not used in the post request. This fixes it and added corresponding test.